### PR TITLE
[#278] Expand documentation with API definitions, message formats and other technical details.

### PR DESCRIPTION
**Pull request by antonia-avramova**
[#278] Expand documentation with API definitions, message formats and other technical details.

Signed-off-by: Antonia Avramova <antonia.avramova@bosch.io>

### CODE REVIEW

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: "Software Updatable API"
+type: docs
+description: >
+  The software updatable service provides the ability to install given list of software modules and to remove modules from containers.
+weight: 4
+---
+
+## **Install**
+Install given list of software modules to the container.
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:10]:
```suggestion
You can install a specified list of containers (software modules).
```

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+> | - | - | - |
+> | topic | `<name>/<namespace>/things/live/messages/install` | - |
+> | path | `/features/SoftwareUpdatable/inbox/messages/install` | - |
+> | **Headers** | | |
+> | response-required | `true` | - |
+> | content-type | `application/json` | - |
+> | correlation-id  | container UUID | - |
+> | **Value** | | |
+> | correlationId | - | different from container UUID |
+> | softwareModules | - | Array from modules, which will be installed |
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:29]:
```suggestion
> | softwareModules | - | An array of modules that will be installed. |
```

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+> | weight | - | - |
+> | metadata | - | Metadata |
+> | forced | true/false | Forced to install the software modules |
+<br>
+
+**Example** : In this example, the User can install listed modules:
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:35]:
```suggestion
**Example** : In this example, you can install the listed modules.
```

File: web/site/content/docs/references/containers/api-reference/software-updatable.md (hunk 2 above)
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:29]: you should describe the software module structure and what parameters it takes, references to the hawk bit APIs would also be helpful

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+				}
+			}
+		]
+	}
+}
+```
+</details>
+
+<details>
+  <summary>Response</summary>
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:81]: the response should follow the format of the request

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+		"correlation-id":"<UUID>"
+	},
+	"path":"/features/SoftwareUpdatable/outbox/messages/install",
+	"status": 204
+}
+```
+</details>
+
+## **Remove**
+Remove software modules from the container.
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:115]:
```suggestion
Remove a software module that is already installed.
```

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+> | Name | Value | Description |
+> | - | - | - |
+> | topic | `<name>/<namespace>/things/live/messages/remove` | - |
+> | path | `/features/SoftwareUpdatable/inbox/messages/remove` | - |
+> | **Headers** | | |
+> | response-required | `true` | - |
+> | content-type | `application/json` | - |
+> | correlation-id  | container UUID | - |
+> | **Value** | | |
+> | value | - | Json presentation of software module to be removed |
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:133]: all parameters of the value should be described

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+				"version":""
+			}
+		]
+	}
+}
+```
+</details>
+
+<details>
+  <summary>Response</summary>
```
Comment (k-gostev) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:163]: the request should follow the format of the response

File: web/site/content/docs/references/file-backup/file-backup.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
+---
+title: "File Backup API"
+type: docs
+description: >
+  The file backup service provides the ability to backup and restore to and from a backend storage.
```
Comment (k-gostev) [inline web/site/content/docs/references/file-backup/file-backup.md:5]:
```suggestion
The file backup service allows you to backup and restore data to and from a backend storage.
```

**Review by k-gostev — CHANGES_REQUESTED** [review]

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 9]
```diff
@@ -0,0 +1,401 @@
...
+> | response-required | true/false | If response is required |
+> | content-type | `application/json` | The content type |
+> | correlation-id | UUID | Used for correlating protocol messages, the same correlation-id as the sent back response message |
+> | **Value** | | |
+> | correlationID | other UUID | Identifier of the triggered file |
+> | **options** | | Options are specific for each provider |
+
+<br>
+
+**Example** : In this example, you can pause an existing and running container.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:118]: This line seems to be copied from CM, fix it accordingly to the FU.

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 10]
```diff
@@ -0,0 +1,401 @@
...
+		"correlation-id":"<UUID>"
+	},
+	"path":"/features/AutoUploadable/outbox/messages/trigger",
+	"status":204
+}
+```
+</details>
+
+## **Cancel**
+Cancel upload the file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:173]:
```suggestion
Cancel a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 11]
```diff
@@ -0,0 +1,401 @@
...
+> | content-type | `application/json` | The content type |
+> | correlation-id | UUID | Used for correlating protocol messages, the same correlation-id as the sent back response message |
+> | **Value** | | |
+> | correlationID | other UUID | Identifier of the uploaded file |
+> | statusCode | | This status code is set to update status code |
+> | message | | This message is set to update status message |
+
+<br>
+
+**Example** : In this example, you can resume cancel operation upload of the file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:197]:
```suggestion
**Example** : In this example, you can cancel a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 12]
```diff
@@ -0,0 +1,401 @@
...
+> | topic | `<name>/<namespace>/things/live/messages/cancel` | Information about the affected Thing and the type of operation |
+> | path | `/features/AutoUploadable/outbox/messages/cancel` | A path that references a part of a Thing which is affected by this message |
+> | **Headers** | | Additional headers |
+> | content-type | `application/json` | The content type |
+> | correlation-id | \<UUID\> | The same correlation id as the sent request message |
+> | **Status** | | Status of the operation cancel upload file |
+
+<br>
+
+**Example** : The response of the cancel upload file operation.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:236]:
```suggestion
**Example** : The response of the cancel file upload operation.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 13]
```diff
@@ -0,0 +1,401 @@
...
+		"correlation-id":"<UUID>"
+	},
+	"path":"/features/AutoUploadable/outbox/messages/cancel",
+	"status":204
+}
+```
+</details>
+
+## **Activate**
+Activate upload of the file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:253]:
```suggestion
Activate a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 14]
```diff
@@ -0,0 +1,401 @@
...
+> | topic | `<name>/<namespace>/things/live/messages/start` | Information about the affected Thing and the type of operation |
+> | path | `/features/AutoUploadable/outbox/messages/start` | A path that references a part of a Thing which is affected by this message |
+> | **Headers** | | Additional headers |
+> | content-type | `application/json` | The content type |
+> | correlation-id | \<UUID\> | The same correlation id as the sent request message |
+> | **Status** | | Status of the operation start upload the file |
+
+<br>
+
+**Example** : The response of the start upload file operation.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:78]:
```suggestion
**Example** : The response of the start file upload operation.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 15]
```diff
@@ -0,0 +1,401 @@
+---
+title: "File Upload API"
+type: docs
+description: >
+  The file upload service provides the ability to start, trigger, cancel, activate or deactivate upload of the file.
+weight: 4
+---
+
+## **Start**
+Start to upload file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:10]:
```suggestion
Start a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 16]
```diff
@@ -0,0 +1,401 @@
...
+		"correlation-id":"<UUID>"
+	},
+	"path":"/features/AutoUploadable/outbox/messages/start",
+	"status": 204
+}
+```
+</details>
+
+## **Trigger**
+Trigger operation is invoked from the backend.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:95]: All operations are invoked by the backend., align with the other descriptions.

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 17]
```diff
@@ -0,0 +1,401 @@
...
+> | response-required | true/false | If response is required |
+> | content-type | `application/json` | The content type |
+> | correlation-id | UUID | Used for correlating protocol messages, the same correlation-id as the sent back response message |
+> | **Value** | | |
+> | from | | A Time after which the upload must be activated |
+> | to | | A Time grater than `from` marks the end of activated |
+
+<br>
+
+**Example** : In this example, you can activate upload of the file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:276]:
```suggestion
**Example** : In this example, you can activate a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 18]
```diff
@@ -0,0 +1,401 @@
...
+> | path | `/features/AutoUploadable/inbox/messages/deactivate` | A path that references a part of a Thing which is affected by this message |
+> | **Headers** | | Additional headers |
+> | response-required | true/false | If response is required |
+> | content-type | `application/json` | The content type |
+> | correlation-id | UUID | Used for correlating protocol messages, the same correlation-id as the sent back response message |
+> | **Value** | | |
+
+<br>
+
+**Example** : In this example, you can deactivate upload of file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:352]:
```suggestion
**Example** : In this example, you can deactivate a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 19]
```diff
@@ -0,0 +1,401 @@
...
+		"correlation-id":"<UUID>"
+	},
+	"path":"/features/AutoUploadable/outbox/messages/activate",
+	"status":204
+}
+```
+</details>
+
+## **Deactivate**
+Deactivate upload file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:331]:
```suggestion
Deactivate a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 20]
```diff
@@ -0,0 +1,401 @@
...
+> | content-type | `application/json` | The content type |
+> | correlation-id | UUID | Used for correlating protocol messages, the same correlation-id as the sent back response message |
+> | **Value** | | |
+> | correlationID | other UUID | Identifier of the uploaded file |
+> | **options** | | Options are specific for each provider |
+> | storage.provider | aws/azure/generic | Storage provider that will be used for uploading the files |
+
+<br>
+
+**Example** : In this example, you can start uploading a file.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:34]:
```suggestion
**Example** : In this example, you can start a file upload.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 21]
```diff
@@ -0,0 +1,401 @@
...
+> | topic | `<name>/<namespace>/things/live/messages/activate` | Information about the affected Thing and the type of operation |
+> | path | `/features/AutoUploadable/outbox/messages/activate` | A path that references a part of a Thing which is affected by this message |
+> | **Headers** | | Additional headers |
+> | content-type | `application/json` | The content type |
+> | correlation-id | \<UUID\> | The same correlation id as the sent request message |
+> | **Status** | | Status of the operation activate upload file |
+
+<br>
+
+**Example** : The response of the activate upload file operation.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:314]:
```suggestion
**Example** : The response of the activate file upload operation.
```

File: web/site/content/docs/references/file-upload/file-upload-api.md [hunk 22]
```diff
@@ -0,0 +1,401 @@
...
+> | topic | `<name>/<namespace>/things/live/messages/deactivate` | Information about the affected Thing and the type of operation |
+> | path | `/features/AutoUploadable/outbox/messages/deactivate` | A path that references a part of a Thing which is affected by this message |
+> | **Headers** | | Additional headers |
+> | content-type | `application/json` | The content type |
+> | correlation-id | \<UUID\> | The same correlation id as the sent request message |
+> | **Status** | | Status of the operation deactivate upload file |
+
+<br>
+
+**Example** : The response of the deactivate upload file operation.
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/file-upload/file-upload-api.md:387]:
```suggestion
**Example** : The response of the deactivate file upload operation.
```

File: web/site/content/docs/references/local-digital-twins/local-digital-twins-config.md
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/local-digital-twins/local-digital-twins-config.md:1]: @k-gostev do we need to move the ldt reference guide in a sub-folder?

Comment (k-gostev) [reply web/site/content/docs/references/local-digital-twins/local-digital-twins-config.md:1]: we can move it in the connectivity folder

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 23]
```diff
@@ -160,10 +160,10 @@ Remove of an installed software module.
 
 **Example** : In this example, you can remove an existing software modules.
 
-**Topic:** `command//edge:device/req//remove`
+**Topic:** `command//edge:device:edge:containers/req//remove`
 ```json
 {
-	"topic":"edge/device/things/live/messages/remove",
+	"topic":"edge/device/things:edge:containers/live/messages/remove",
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:166]:
```suggestion
	"topic":"edge/device:edge:containers/things/live/messages/remove",
```

File: web/site/content/docs/references/containers/api-reference/software-updatable.md [hunk 24]
```diff
@@ -116,7 +116,7 @@ You can install a specified list of containers (software modules).
 **Topic:** `command//edge:device/res//install``
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/containers/api-reference/software-updatable.md:116]:
```suggestion
**Topic:** `command//edge:device:edge:containers/res//install``
```

File: web/site/content/docs/references/containers/api-reference/container.md [hunk 25]
```diff
@@ -223,7 +223,7 @@ Stop an existing and running container with given options.
 **Topic:** `command//edge:device/res//stopWithOptions``
```
Comment (dimitar-dimitrow) [inline web/site/content/docs/references/containers/api-reference/container.md:223]:
```suggestion
**Topic:** `command//edge:device:ege:containers/res//stopWithOptions``
```

**Review by dimitar-dimitrow — APPROVED** [review]

**Pull request merged**